### PR TITLE
Add MODSEC_ARGS_COMBINED_SIZE to set tx.total_arg_length

### PR DESCRIPTION
**Pull request by soufianebenali**
Adds ENV variable to set  `tx.total_arg_length`.
@franbuehler please review.

Resolves:  [FILO-328](https://control.vshn.net/tickets/FILO-328)

### CODE REVIEW

File: v3.1/custom-rules/before-crs.dist/baseconfig.conf [hunk 1]
```diff
@@ -7,9 +7,10 @@ SecAction "id:900110,phase:1,pass,nolog,\
 SecAction "id:900120,phase:1,pass,nolog,\
   setvar:tx.paranoia_level=${PARANOIA}"
 
-SecAction "id:900200,phase:1,pass,nolog,setvar:'tx.allowed_methods=${MODSEC_ALLOWED_METHODS}'"
-SecAction "id:900220,phase:1,pass,nolog,setvar:'tx.allowed_request_content_type=${MODSEC_ALLOWED_CONTENT}'"
+SecAction "id:900200,phase:1,pass,nolog,setvar:tx.allowed_methods=${MODSEC_ALLOWED_METHODS}"
+SecAction "id:900220,phase:1,pass,nolog,setvar:tx.allowed_request_content_type=${MODSEC_ALLOWED_CONTENT}"
```
Comment (bittner) [inline v3.1/custom-rules/before-crs.dist/baseconfig.conf]: @franbuehler Why is this important to leave it like this? It looks erroneous.

**Comment by bittner** [comment]
> And please be sure, that the uncommented part of crs-setup.conf inside the image is commented out again.

What is the "uncommented part"? I'm not sure I understand.

**Comment by bittner** [comment]
Sorry about the wrong rule ID, we should have figured that. It all makes sense now.

The rule number sequence in our `baseconfig.conf` has gaps, though. Is there a full list of all IDs in use somewhere? I believe, we looked at the original `owasp-crs/crs-setup.conf.example` file in the image and didn't find those rules.

**Comment by franbuehler** [comment]
> > And please be sure, that the uncommented part of crs-setup.conf inside the image is commented out again.
> 
> Do you intend what is in `owasp-crs/crs-setup.conf` in the image?
> 
> ```
> # grep -A 5 -B 2 900330 /etc/apache2/modsecurity.d/owasp-crs/crs-setup.conf
> # Uncomment this rule to set a limit.
> SecAction \
>  "id:900330, \
>  phase:1, \
>  nolog, \
>  pass, \
>  t:none, \
>  setvar:tx.total_arg_length=64000"
> ```

Yes, that's the file in the image I mean.

**Comment by franbuehler** [comment]
> Sorry about the wrong rule ID, we should have figured that. It all makes sense now.
> 
> The rule number sequence in our `baseconfig.conf` has gaps, though. Is there a full list of all IDs in use somewhere? I believe, we looked at the original `owasp-crs/crs-setup.conf.example` file in the image and didn't find those rules.

No problem.
I took the same id as found in: https://github.com/SpiderLabs/owasp-modsecurity-crs/blob/v3.2.0/crs-setup.conf.example

**Comment by bittner** [comment]
I would have hoped we're overriding this, but the current change is in [before-crs.dist](https://github.com/vshn/modsecurity-docker/tree/master/v3.1/custom-rules/before-crs.dist) and the default `crs-setup.conf` is [loaded afterwards](https://github.com/vshn/modsecurity-docker/blob/master/v3.1/modsecurity.d/include.conf#L5-L8). That's unfortunate, creates additional work.

We'll try to incorporate those environment variables in the [base image](https://github.com/CRS-support/modsecurity-crs-docker) in future. That will make things easier then. :worried:

**Comment by franbuehler** [comment]
Thanks a lot. So, I'll wait for your changes.

**Comment by bittner** [comment]
If I see this correctly we can actually remove (or not include) the `crs-setup.conf` file entirely. Except for the `tx.crs_setup_version` variable there are no values that we don't already set anyway:

```apache
# sed -e '/^#/d' -e '/^$/d' /etc/apache2/modsecurity.d/owasp-crs/crs-setup.conf
SecDefaultAction "phase:1,log,auditlog,pass"
SecDefaultAction "phase:2,log,auditlog,pass"
SecAction \
  "id:900000, \
   phase:1, \
   nolog, \
   pass, \
   t:none, \
   setvar:tx.paranoia_level=1"
SecAction \
 "id:900330, \
 phase:1, \
 nolog, \
 pass, \
 t:none, \
 setvar:tx.total_arg_length=64000"
SecCollectionTimeout 600
SecAction \
 "id:900990,\
  phase:1,\
  nolog,\
  pass,\
  t:none,\
  setvar:tx.crs_setup_version=310"
```

Or do you think it's still worth including that file in any case, @franbuehler?

**Comment by bittner** [comment]
Even worse: The `crs-setup.conf` is enforcing the `tx.paranoia_level` to stay `1`. That must be fixed, too.

**Comment by bittner** [comment]
As explained in the commit:

Only the CRS version value is included from the `crs-setup.conf` with the above change from now on. We should already cover all other values via the environment variables we have introduced, if I'm not mistaken.

**Comment by bittner** [comment]
Reading the example configuration file again, I just figured that we probably removed a value the _needs_ to stay in the `crs-setup.conf` file:

```apache
#
# -- [[ Collection timeout ]] --------------------------------------------------
#
# Set the SecCollectionTimeout directive from the ModSecurity default (1 hour)
# to a lower setting which is appropriate to most sites.
# This increases performance by cleaning out stale collection (block) entries.
#
# This value should be greater than or equal to:
# tx.reput_block_duration (see section "Blocking Based on IP Reputation") and
# tx.dos_block_timeout (see section "Anti-Automation / DoS Protection").
#
# Ref: https://github.com/SpiderLabs/ModSecurity/wiki/Reference-Manual#wiki-SecCollectionTimeout

# Please keep this directive uncommented.
# Default: 600 (10 minutes)
SecCollectionTimeout 600
```

This is right above the `tx.crs_setup_version` `SecAction` (in older versions of the file there was a `SecRule` to enable "Debug Mode" between that and the latter). If a fix for this is needed it would be about [adapting the grep](https://github.com/vshn/modsecurity-docker/blob/master/v3.1/Dockerfile#L84-L86) as follows:
```console
grep -B 15 -e setvar:tx.crs_setup_version -e SecCollectionTimeout \
      /opt/owasp-modsecurity-crs/crs-setup.conf | sed 's/^--$//' > \
      /etc/apache2/modsecurity.d/owasp-crs/crs-setup.conf
```
Alternatively, would it maybe make sense to add an environment variable for that?

Cc @srueg

**Comment by bittner** [comment]
I pushed a fix for the above to PR #44.

**Pull request merged**